src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/CourseActivityRepository.kt

```diff
@@ -26,16 +26,13 @@ interface CourseActivityRepository : JpaRepository<CourseActivity, Long> {
       (
        select distinct(opp.courseActivity.courseActivityId) 
        from OffenderProgramProfile opp
-       join OffenderBooking ob on opp.offenderBooking = ob
        where opp.prison.id = :prisonId 
-       and opp.programStatus.code = 'ALLOC'
-       and (opp.endDate is null or opp.endDate > current_date)
-       and ob.active = true
-       and ob.location.id = :prisonId
+       and opp.startDate is not null
+       and (opp.endDate is null or opp.endDate >= :oldestAllocation)
       )   
   """,
   )
-  fun findActiveActivities(prisonId: String, courseActivityId: Long?, pageRequest: Pageable): Page<Long>
+  fun findActiveActivities(prisonId: String, courseActivityId: Long?, pageRequest: Pageable, oldestAllocation: LocalDate = LocalDate.now().minusMonths(6)): Page<Long>
```

Comment:
There is no JPQL syntax for checking a date is at least 6 months ago - therefore we have to use a parameter.